src/thor/dijkstras.cc

```diff
@@ -354,7 +357,10 @@ void Dijkstras::Compute(google::protobuf::RepeatedPtrField<valhalla::Location>&
 
     const baldr::DirectedEdge* opp_pred_edge = nullptr;
     if (expansion_direction == ExpansionType::reverse) {
-      opp_pred_edge = graphreader.GetGraphTile(pred.opp_edgeid())->directededge(pred.opp_edgeid());
+      const auto opp_edge_tile = graphreader.GetGraphTile(pred.opp_edgeid());
+      if (opp_edge_tile != nullptr) {
+        opp_pred_edge = opp_edge_tile->directededge(pred.opp_edgeid());
+      }
```

Comment:
we should fail here as soon as we know we cant get the tile, not wait until later

```suggestion
      opp_pred_edge = graphreader.GetOpposingEdge(pred.opp_edgeid());
      if (opp_edge_tile != nullptr) {
        continue;
      }
```

Reply:
https://github.com/valhalla/valhalla/pull/3065#discussion_r637948351 refers to this suggestion